File: varianti/Dosrok_2026/24.py
s = open('24_28765.txt')
s = s.readline()

c = ''
m = 0
k = 0

for i in range(len(s)):
    c += s[i]
    if c[-2:] == 'BC':
        k += 1
    while k > 180:
        if c[:2] == 'BC':
            k -= 1
        c = c[1:]
    m = max(m, len(c))
    if i % 100000 == 0:
        print(i, len(s), m)
print(m)

File: varianti/Dosrok_2026/24.cpp
#include <iostream>                                                                 
#include <fstream>
#include <string>
using namespace std;

int maxSequenceWithLimitedPairs(const string &filename) {
    ifstream file(filename);
    
    if (!file.is_open()) {
        cerr << endl;
        return -1;
    }

    int max_length = 0;
    int left = 0;
    int bc_count = 0;

    string line;
    while (getline(file, line)) {
        for (size_t right = 0; right < line.length(); ++right) {
            if (line[right] == 'B' && right + 1 < line.length() && line[right + 1] == 'C') {
                bc_count++;
            }
            
            while (bc_count > 180) {
                if (left < line.length() - 1 && line[left] == 'B' && line[left + 1] == 'C') {
                    bc_count--;
                }
                left++;
            }
            
            max_length = max(max_length, static_cast<int>(right - left + 1));
        }
    }

    file.close();
    return max_length;
}

int main() {
    const string filename = 'input.txt';
    int result = maxSequenceWithLimitedPairs(filename);
    
    cout << 'Максимальная длина:' << result << endl;
    return 0;
}